test/nettests/dash.cpp

```diff
@@ -12,7 +12,7 @@ using namespace mk::nettests;
 #ifdef ENABLE_INTEGRATION_TESTS
 
 TEST_CASE("Synchronous dash test") {
-    test::nettests::make_test<DashTest>().run();
+    test::nettests::with_test<DashTest>(test::nettests::run_test);
```

Comment:
From here onwards: I needed to rewrite some tests to make sure we wait for background threads to terminate, so that the thread local storage associated with them has been cleared. Otherwise Valgrind complains.